pandas/plotting/_core.py

```diff
@@ -370,1305 +642,760 @@
-            Allows plotting of one column versus another. If not specified,
-            all numerical columns are used.
-        **kwds
-            Additional keyword arguments are documented in
-            :meth:`DataFrame.plot`.
 
-        Returns
-        -------
-        matplotlib.axes.Axes or np.ndarray of them
-            An ndarray is returned with one :class:`matplotlib.axes.Axes`
-            per column when ``subplots=True``.
+def hist_series(self, by=None, ax=None, grid=True, xlabelsize=None,
+                xrot=None, ylabelsize=None, yrot=None, figsize=None,
+                bins=10, **kwds):
+    """
+    Draw histogram of the input series using matplotlib.
 
-        See Also
-        --------
-        DataFrame.plot.barh : Horizontal bar plot.
-        DataFrame.plot : Make plots of a DataFrame.
-        matplotlib.pyplot.bar : Make a bar plot with matplotlib.
+    Parameters
+    ----------
+    by : object, optional
+        If passed, then used to form histograms for separate groups
+    ax : matplotlib axis object
+        If not passed, uses gca()
+    grid : bool, default True
+        Whether to show axis grid lines
+    xlabelsize : int, default None
+        If specified changes the x-axis label size
+    xrot : float, default None
+        rotation of x axis labels
+    ylabelsize : int, default None
+        If specified changes the y-axis label size
+    yrot : float, default None
+        rotation of y axis labels
+    figsize : tuple, default None
+        figure size in inches by default
+    bins : integer or sequence, default 10
+        Number of histogram bins to be used. If an integer is given, bins + 1
+        bin edges are calculated and returned. If bins is a sequence, gives
+        bin edges, including left edge of first bin and right edge of last
+        bin. In this case, bins is returned unmodified.
+    `**kwds` : keywords
+        To be passed to the actual plotting function
 
-        Examples
-        --------
-        Basic plot.
+    Returns
+    -------
+    matplotlib.AxesSubplot
+        A histogram plot.
 
-        .. plot::
-            :context: close-figs
+    See Also
+    --------
+    matplotlib.axes.Axes.hist : Plot a histogram using matplotlib.
+    """
+    plot_backend = _get_plot_backend()
+    return plot_backend.hist_series(self, by=by, ax=ax, grid=grid,
+                                    xlabelsize=xlabelsize, xrot=xrot,
+                                    ylabelsize=ylabelsize, yrot=yrot,
+                                    figsize=figsize, bins=bins, **kwds)
 
-            >>> df = pd.DataFrame({'lab':['A', 'B', 'C'], 'val':[10, 30, 20]})
-            >>> ax = df.plot.bar(x='lab', y='val', rot=0)
 
-        Plot a whole dataframe to a bar plot. Each column is assigned a
-        distinct color, and each row is nested in a group along the
-        horizontal axis.
+def hist_frame(data, column=None, by=None, grid=True, xlabelsize=None,
```

Comment:
It makes sense to move those non-accessor functions down in the file, but for the (already large) diff it might make sense to keep them at the top where they were (those ones are just a copy paste I think), and only do the move in a follow-up PR

Reply:
Yes, that makes sense, but surely better in a follow up.

Reply:
@datapythonista I actually meant moving them back here, so we can do the actual move in a follow-up 
(the diff is right now basically not informative)